cpp/src/arrow/compare.cc

```diff
@@ -700,6 +734,15 @@ class TypeEqualsVisitor {
     return Status::OK();
   }
 
+  Status Visit(const RunEndEncodedType& left) {
+    const auto& right = checked_cast<const RunEndEncodedType&>(right_);
+    // NOTE(felipecrv): the type of the run_ends might differ, but that doesn't
+    // change the logical meaning of the type so only the encoded type is
+    // compared.
```

Comment:
That's inconsistent with how dictionary equality works (see above), and would also break `CompareRunEndEncoded` above.

Reply:
Thanks. I will make it more strict.